pulse-opl/src/delay.rs

```diff
@@ -94,22 +104,15 @@ pub struct Delay {
 impl_dyn_hash!(Delay);
 
 impl Delay {
-    pub fn new(axis: usize, input_fact: &TypedFact, delay: usize, overlap: usize) -> Delay {
-        let axis = axis;
-        let mut buffer_shape = input_fact.shape.to_tvec();
-        buffer_shape[axis] = (delay + overlap).to_dim();
-        Delay { datum_type: input_fact.datum_type, buffer_shape, axis, delay, overlap }
-    }
-
     pub fn new_typed(
         input_fact: &TypedFact,
         axis: usize,
         delay: usize,
         overlap: usize,
-    ) -> TractResult<Delay> {
+    ) -> Delay {
```

Comment:
mmm... changing this return type may break a few places in NNEF and pulse, i think... not a problem, this API is considered private, but we need to fix them.

Reply:
Indeed, I fixed the different places where it was used. The `new` API was returning directly `Delay` without Result. I remove this API that was a duplicate with `new_typed`.